processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -95,9 +107,7 @@ func New(
 	associations []Association,
 	exclude Excludes,
 	newClientSet APIClientsetProvider,
-	newInformer InformerProvider,
-	newNamespaceInformer InformerProviderNamespace,
-	newReplicaSetInformer InformerProviderReplicaSet,
+	informersFactory InformersFactoryList,
```

Comment:
small refactoring for potential future extensions

Reply:
Usually we have refactorings in standalone PRs first :): https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#refactoring-work.

But I would be fine keeping this here if others agree.